core/templates/components/summary-tile/topic-summary-tile.component.html

```diff
@@ -11,7 +11,7 @@
       </picture>
     </div>
     <div class="topic-details" [ngStyle]="{'background-color': getDarkerThumbnailBgColor()}" ng-class="{'mask-effect': !isPublished}">
-      <span class="topic-name">{{ topicSummary.getName() }}</span>
+      <span class="topic-name">{{ topicNameTranslationKey | translate }}</span>
```

Comment:
I think this should use the translation key when the language is not English, but the topic summary's name when it is English.

That way, if we ever change a topic name, the English version will at least update (and then the other translations can be updated on translatewiki and brought back into the system later).

You might need to implement a method/service that encapsulates this logic.

Reply:
So I modified it to be implemented as you said, just that the value of `topicNameTranslationKey` stays `topicSummary.getName()` if the language is english, this functionality is handled in translation service method which is used to get the value of  `topicNameTranslationKey`

Reply:
I think you need to implement it in such a way that the names accurately describe the objects. You can't use the name topicNameTranslationKey to refer to the name of the topic, because semantically those don't mean the same thing.

Differentiate between the translation keys and e.g. topicNameToDisplay (say). In the code, every variable name must accurately represent what the variable contains and there shouldn't be any cases where one thing masquerades as another thing -- otherwise this will really confuse developers.